checker/src/org/checkerframework/checker/guieffect/GuiEffectVisitor.java

```diff
@@ -65,14 +66,102 @@ protected void checkMethodInvocability(
         // TODO: Undo this hack!
     }
 
+    protected class GuiEffectOverrideChecker extends OverrideChecker {
+        /**
+         * Extend the receiver part of the method override check. We extend the standard check, to
+         * additionally permit narrowing the receiver's permission to <code>@AlwaysSafe</code> in a
+         * safe instantiation of a <code>@PolyUIType</code> Returns true if the override is
+         * permitted.
+         */
+        @Override
+        protected boolean checkReceiverOverride() {
+            // We cannot reuse the inherited method because it directly issues the failure, but we want a more permissive check.  So this is copied down and modified from BaseTypeVisitor.OverrideChecker.checkReceiverOverride.
```

Comment:
Please wrap comments to either 80 or 100 columns (your choice).

(GitHub brokenly refuses to show comments longer than that!  But it's mostly for consistency with the rest of the codebase, and readability.)